netprofile_mailing/netprofile_mailing/models.py

```diff
@@ -0,0 +1,455 @@
+	print_function,
+	absolute_import,
+	division
+)
+
+__all__ = [
+	'MailingTemplate',
+	'MailingLog'
+	'MailingSubscription',
+]
+
+import hashlib
+import datetime
+import smtplib
+import json 
+import transaction
+
+from pyramid.threadlocal import get_current_registry
+
+from sqlalchemy import (
+	Column,
+	DateTime,
+	FetchedValue,
+	ForeignKey,
+	Index,
+	Sequence,
+	TIMESTAMP,
+	Unicode,
+	UnicodeText
+)
+
+from sqlalchemy.orm import (
+	backref,
+	relationship
+)
+
+from sqlalchemy.ext.associationproxy import association_proxy
+
+from sqlalchemy.ext.hybrid import hybrid_property
+
+from netprofile.db.connection import (
+	Base,
+	DBSession
+)
+from netprofile.db.fields import (
+	ASCIIString,
+	NPBoolean,
+	UInt8,
+	UInt16,
+	UInt32,
+	npbool,
+	IPv4Address,
+	IPv6Address,
+	IPv6Offset
+)
+from netprofile.db.ddl import (
+	Comment,
+	Trigger
+)
+from netprofile.tpl import TemplateObject
+from netprofile.ext.columns import MarkupColumn
+from netprofile.common.hooks import register_hook
+from netprofile.ext.data import (
+	ExtModel,
+	_name_to_class
+)
+from netprofile.ext.wizards import (
+	ExternalWizardField,
+	SimpleWizard,
+	Step,
+	Wizard
+)
+from pyramid.i18n import (
+	TranslationStringFactory,
+	get_localizer
+)
+
+from netprofile_access.models import AccessEntity
+
+_ = TranslationStringFactory('netprofile_mailing')
+
+@register_hook('np.wizard.init.mailing.MailingLog')
+def _wizcb_maillog_submit(wiz, step, act, val, req):
+	sess = DBSession()
+	cfg = get_current_registry().settings
+	userIDs = json.loads(val['userid'])
+	userlist = val['user']
+	templateName = val['template']
+	templId = val['templid']
+	receiver = None
+	sender = cfg.get('netprofile.mailing.sender', 'admin@mysite.com')
+	sendername = cfg.get('sender.name', 'localadmin')
+	mailhost = cfg.get('mail.host', 'localhost')
+
+	for userid in userIDs:
+		resvalue = {'userid' : userid}
+		user = sess.query(AccessEntity).filter(AccessEntity.id==userid).first()
+		subscr = sess.query(MailingSubscription).filter(MailingSubscription.userid==userid).first()
+		print("########################## SUBSCR ###########################")
+		
+		#a long try-except statement to check if user is in mailing list
+		try:
+			if subscr.issubscribed is True:
+				templateBody = sess.query(MailingTemplate).filter(MailingTemplate.id==templId).first().body
+				resvalue['user'] = user
+				resvalue['template'] = templateBody
+				resvalue['templid'] = templId
+		
+				if user.parent:
+					try:
+						receiver = user.parent.email
+					except AttributeError:
+						print("################### USER'S PARENT HAVE NO EMAIL ATTRIBUTE #######################")
+	
+				if receiver is not None:
+					message = """From: From {0} <{1}>
+To: {2} <{3}>
+MIME-Version: 1.0
+Content-type: text/html
+Subject: {4}
+
+{5}
+""".format(sendername, sender, user.nick, receiver,  templateName, templateBody)
+					
+					smtpObj = smtplib.SMTP(mailhost)
+					smtpObj.sendmail(sender, receiver, message)         
```

Comment:
You should send mail by pyramid_mailer, like here
https://github.com/unikmhz/npui/blob/master/netprofile_access/netprofile_access/views.py#L409